## Studies stopped because of efficacy

```{r, efficacy_runFisherTests, cache = TRUE}
aggregations_path <- "gs://ot-team/dochoa/predictions_aggregations/*.parquet"
all_aggregations <- spark_read_parquet(sc, aggregations_path) %>%
    mutate(b = predictionTotal - a) %>%
    mutate(c = comparisonTotal - a) %>%
    mutate(d = total - b - c + a) %>%
    collect()

all_tests <- bind_cols(
    all_aggregations,
    all_aggregations %>%
        mutate(d = ifelse(d < 0, 0, d)) %>%
        rowwise() %>%
        do(broom::tidy(fisher.test(matrix(c(.$a, .$b, .$c, .$d), nrow = 2))))
)
```

### Stop reasons by datatype

```{r, efficacy_main, cache = TRUE}
stars_pval <- function(p_value) {
  unclass(
    symnum(
      p_value,
      corr = FALSE,
      na = FALSE,
      cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05,  1),
      symbols = c("****", "***", "**", "*",  "ns")
    )
  )
}

toplot_efficacy_main <- all_tests %>%
    filter(predictionType %in% c("metareason", "clinical", "stopStatus")) %>%
    filter(comparison %in% c("animal_model", "genetic_association")) %>%
    filter(comparisonType == "byDatatype") %>%
    mutate(predictionType = fct_relevel(
        predictionType,
        "clinical",
        "stopStatus",
        "metareason"
    )) %>%
    mutate(predictionType = fct_recode(predictionType,
        "All studies" = "clinical",
        "Stopped studies" = "stopStatus",
        "Reason implication" = "metareason"
    )) %>%
    mutate(prediction = fct_reorder(prediction, estimate, median)) %>%
    mutate(prediction = fct_recode(prediction,
        "Suspicious" = "Business_Administrative",
        "Invalid" = "Invalid_Reason",
        "Safety" = "Safety_Sideeffects",
        "Negative" = "Negative",
        "Neutral" = "Neutral",
        "Success" = "Success"
    )) %>%
    mutate(prediction = fct_relevel(prediction, "Phase IV", after = Inf)) %>%
    mutate(oddsinterval = paste(
        round(estimate, digits = 2),
        " (",
        round(conf.low, digits = 2),
        "-",
        round(conf.high, digits = 2),
        ")",
    sep = "")) %>%
    mutate(significance = stars_pval(p.value))

p_efficacy_main_dots <- toplot_efficacy_main %>%
    filter(comparison == "genetic_association") %>%
    ggplot(aes(x = prediction, y = estimate)) +
    geom_hline(aes(yintercept = 1),
        size = .25,
        linetype = "dashed"
    ) +
    geom_errorbar(
        aes(ymin = conf.low, ymax = conf.high),
        width = 0,
        color = "steelblue",
        size = 1) +
    geom_point(color = "steelblue", size = 3) +
    scale_y_log10(name = "Odds Ratio (log)", position = "right") +
    scale_x_discrete(labels = function(x)
    stringr::str_wrap(x, width = 25)) +
    coord_flip() +
    facet_grid(predictionType ~ .,
        scale = "free_y",
        space = "free_y",
        switch = "y"
    ) +
    theme_cowplot(font_size = 12) +
    panel_border(color = "black") +
    theme(
        axis.title.y = element_blank(),
        # strip.text.y = element_blank(),
        axis.line = element_blank(),
        # panel.spacing = unit(0, "lines"),
        panel.spacing = unit(-0.1, "lines"),
        strip.placement = "outside",
        strip.background = element_blank(),
        strip.switch.pad.grid = unit(20, "pt")
    )

p_efficacy_main_numbers <- toplot_efficacy_main %>%
    filter(comparison == "genetic_association") %>%
    mutate(a = formatC(a, format = "d", big.mark = ",")) %>%
    select(
        predictionType,
        prediction,
        a,
        oddsinterval,
        significance
    ) %>%
    gather(key, value, -predictionType, -prediction) %>%
    mutate(key = fct_recode(
        key,
        "n" = "a",
        "OR (95% CI)" = "oddsinterval",
        "Significance" = "significance"
    )) %>%
    ggplot(aes(x = prediction, y = key)) +
    # geom_tile(fill = NA, color = "black") +
    geom_tile(fill = NA) +
    geom_text(aes(label = value)) +
    scale_x_discrete(labels = function(x) stringr::str_wrap(x, width = 25)) +
    scale_y_discrete(expand = c(0, 0), position = "right") +
    coord_flip() +
    facet_grid(predictionType ~ .,
        scale = "free_y",
        space = "free_y"
    ) +
    theme_cowplot(font_size = 12) +
    panel_border(color = "black") +
    theme(
        axis.ticks = element_blank(),
        axis.title = element_blank(),
        axis.text.y = element_blank(),
        panel.spacing = unit(-0.1, "lines"),
        strip.background = element_blank(),
        axis.line = element_blank(),
        strip.text.y = element_blank()
    )

p_efficacy_main <- plot_grid(
    p_efficacy_main_dots +
    theme(plot.margin = margin(
        t = 5.5, b = 5.5, l = 5.5, r = -4, unit = "pt")),
    p_efficacy_main_numbers +
    theme(plot.margin = margin(
        t = 5.5, b = 5.5, l = 0, r = 5.5, unit = "pt")),
    rel_widths = c(1, 0.9),
    align = "h",
    axis = "t",
    nrow = 1,
    ncol = 2
)
p_efficacy_main

p_animal_main_dots <- toplot_efficacy_main %>%
    filter(comparison == "animal_model") %>%
    filter(prediction != "Success") %>%
    ggplot(aes(x = prediction, y = estimate)) +
    geom_hline(aes(yintercept = 1),
        size = .25,
        linetype = "dashed"
    ) +
    geom_errorbar(
        aes(ymin = conf.low, ymax = conf.high),
        width = 0,
        color = "steelblue",
        size = 1) +
    geom_point(color = "steelblue", size = 3) +
    scale_y_log10(name = "Odds Ratio (log)", position = "right") +
    scale_x_discrete(labels = function(x)
    stringr::str_wrap(x, width = 25)) +
    coord_flip() +
    facet_grid(predictionType ~ .,
        scale = "free_y",
        space = "free_y",
        switch = "y"
    ) +
    theme_cowplot(font_size = 12) +
    panel_border(color = "black") +
    theme(
        #Removing axis
        axis.ticks = element_blank(),
        axis.title.y = element_blank(),
        axis.text.y = element_blank(),
        panel.spacing = unit(-0.1, "lines"),
        strip.background = element_blank(),
        axis.line = element_blank(),
        strip.text.y = element_blank()

    )

p_animal_main_numbers <- toplot_efficacy_main %>%
    filter(comparison == "animal_model") %>%
    filter(prediction != "Success") %>%
    mutate(a = formatC(a, format = "d", big.mark = ",")) %>%
    select(
        predictionType,
        prediction,
        a,
        oddsinterval,
        significance
    ) %>%
    gather(key, value, -predictionType, -prediction) %>%
    mutate(key = fct_recode(
        key,
        "n" = "a",
        "OR (95% CI)" = "oddsinterval",
        "Significance" = "significance"
    )) %>%
    ggplot(aes(x = prediction, y = key)) +
    # geom_tile(fill = NA, color = "black") +
    geom_tile(fill = NA) +
    geom_text(aes(label = value)) +
    scale_x_discrete(labels = function(x) stringr::str_wrap(x, width = 25)) +
    scale_y_discrete(expand = c(0, 0), position = "right") +
    coord_flip() +
    facet_grid(predictionType ~ .,
        scale = "free_y",
        space = "free_y"
    ) +
    theme_cowplot(font_size = 12) +
    panel_border(color = "black") +
    theme(
        axis.ticks = element_blank(),
        axis.title = element_blank(),
        axis.text.y = element_blank(),
        panel.spacing = unit(-0.1, "lines"),
        strip.background = element_blank(),
        axis.line = element_blank(),
        strip.text.y = element_blank()
    )

p_animal_main <- plot_grid(
    p_animal_main_dots +
    theme(plot.margin = margin(
        t = 5.5, b = 5.5, l = 5.5, r = -4, unit = "pt")),
    p_animal_main_numbers +
    theme(plot.margin = margin(
        t = 5.5, b = 5.5, l = 0, r = 5.5, unit = "pt")),
    rel_widths = c(1, 1.12),
    align = "h",
    axis = "t",
    nrow = 1,
    ncol = 2
)
p_animal_main


```

```{r, efficacy_meta_stop_by_datatype, cache = TRUE}

p_meta_stop_by_datatype <- all_tests %>%
    filter(predictionType %in% c("metareason", "clinical", "stopStatus")) %>%
    filter(comparison %in% c("animal_model", "genetic_association")) %>%
    filter(comparisonType == "byDatatype") %>%
    mutate(predictionType = fct_relevel(
        predictionType,
        "clinical",
        "stopStatus",
        "metareason")) %>%
    mutate(prediction = fct_reorder(prediction, estimate, median)) %>%
    mutate(comparison = fct_rev(comparison)) %>%
    ggplot(aes(x = prediction, y = estimate)) +
    geom_hline(aes(yintercept = 1),
        size = .25,
        linetype = "dashed"
    ) +
    geom_errorbar(aes(ymin = conf.low, ymax = conf.high),
    width = 0) +
    geom_point() +
    scale_y_log10(name = "Odds Ratio (log)") +
    coord_flip() +
    facet_grid(
        predictionType ~ comparison,
        scales = "free_y",
        space = "free_y") +
    theme_cowplot(font_size = 12) +
    panel_border(color = "black") +
    theme(
        axis.title.y = element_blank(),
        # strip.text.y = element_blank(),
        strip.background = element_rect(fill = NA),
        panel.grid.major.y = element_blank()
    )
p_meta_stop_by_datatype
```

### More granular stop reasons by datatype

```{r, efficacy_meta_stop_by_datatype_granular, cache = TRUE}
p_meta_stop_by_dt_granular <- all_tests %>%
    filter(predictionType %in% c("reason", "clinical", "stopStatus")) %>%
    filter(comparison %in% c("animal_model", "genetic_association")) %>%
    filter(comparisonType == "byDatatype") %>%
    mutate(predictionType = fct_relevel(
        predictionType,
        "clinical",
        "stopStatus",
        "reason"
    )) %>%
    mutate(predictionType = fct_recode(
        predictionType,
        "All studies" = "clinical",
        "Stopped studies" = "stopStatus",
        "Reason" = "reason"
    )) %>%
    mutate(prediction = str_replace_all(prediction, "_", " ")) %>%
    mutate(prediction = str_to_sentence(prediction)) %>%
    mutate(prediction = fct_recode(
        prediction,
        "Phase IV" = "Phase iv",
        "Phase III+" = "Phase iii+",
        "Phase II+" = "Phase ii+"
    )) %>%
    mutate(prediction = fct_reorder(prediction, estimate, median)) %>%
    mutate(prediction = fct_relevel(prediction, "Phase IV", after = Inf)) %>%
    mutate(comparison = fct_recode(
        comparison,
        "Genetic association" = "genetic_association",
        "Mouse KO" = "animal_model"
    )) %>%
    mutate(comparison = fct_rev(comparison)) %>%
    ggplot(aes(x = prediction, y = estimate)) +
    geom_hline(aes(yintercept = 1),
        size = .25,
        linetype = "dashed"
    ) +
    geom_errorbar(aes(ymin = conf.low, ymax = conf.high),
    width = 0) +
    geom_point() +
    scale_y_log10(name = "Odds Ratio") +
    coord_flip() +
    facet_grid(
        predictionType ~ comparison,
        scales = "free_y",
        space = "free_y") +
    theme_cowplot(font_size = 12) +
    panel_border(color = "black") +
    theme(
        axis.title.y = element_blank(),
        # strip.text.y = element_blank(),
        strip.background = element_rect(fill = NA),
        panel.grid.major.y = element_blank()
    )
p_meta_stop_by_dt_granular

```


### Stop reasons (meta) by genetic datasource

```{r, efficacy_meta_by_genetic_ds, cache = TRUE}
genetic_ds <- c(
    "clingen",
    "eva",
    # "gene2phenotype",
    "genomics_england",
    "ot_genetics_portal",
    # "phewas_catalog",
    "uniprot_literature",
    "uniprot_variants"
)
p_efficacy_meta_by_genetic_ds <- all_tests %>%
    filter(predictionType %in% c("metareason", "clinical", "stopStatus")) %>%
    filter(comparison %in% genetic_ds) %>%
    filter(comparisonType == "byDatasource") %>%
    mutate(predictionType = fct_relevel(
        predictionType,
        "clinical", "stopStatus", "metareason"
    )) %>%
    mutate(predictionType = fct_recode(
        predictionType,
        "All studies" = "clinical",
        "Stopped studies" = "stopStatus",
        "Reason implication" = "metareason"
    )) %>%
    mutate(prediction = fct_recode(prediction,
        "Suspicious" = "Business_Administrative",
        "Invalid" = "Invalid_Reason",
        "Safety" = "Safety_Sideeffects",
        "Negative" = "Negative",
        "Neutral" = "Neutral"
    )) %>%
    mutate(prediction = fct_reorder(prediction, estimate, median)) %>%
    mutate(prediction = fct_relevel(prediction, "Phase IV", after = Inf)) %>%
    mutate(comparison = fct_relevel(
        comparison,
        "ot_genetics_portal",
        "eva",
        "genomics_england",
        "clingen",
        "uniprot_variants",
        "uniprot_literature")) %>%
    mutate(comparison = fct_recode(
        comparison,
        "ClinVar" = "eva",
        "Clingen (Gene validity)" = "clingen",
        "GEL PanelApp" = "genomics_england",
        "OT Genetics Portal" = "ot_genetics_portal",
        "Uniprot (variants)" = "uniprot_literature",
        "Uniprot (gene-disease)" = "uniprot_variants"
    )) %>%
    ggplot(aes(x = prediction, y = estimate)) +
    geom_hline(aes(yintercept = 1),
        size = .25,
        linetype = "dashed"
    ) +
    geom_errorbar(aes(ymin = conf.low, ymax = conf.high),
    width = 0) +
    geom_point() +
    scale_y_log10(name = "Odds Ratio") +
    coord_flip() +
    facet_grid(
        predictionType ~ comparison,
        scales = "free_y",
        space = "free_y") +
    theme_cowplot(font_size = 12) +
    panel_border(color = "black") +
    theme(
        axis.title.y = element_blank(),
        # strip.text.y = element_blank(),
        strip.background = element_rect(fill = NA),
        panel.grid.major.y = element_blank()
    )
p_efficacy_meta_by_genetic_ds

```


### GWAS causality

```{r, efficacy_gwasL2Gscore, cache = TRUE}
p_efficacy_gwas_l2g_score <- all_tests %>%
    filter(comparisonType == "l2g") %>%
    filter(prediction %in% c("Phase III+", "Negative")) %>%
    mutate(prediction = fct_recode(
        prediction,
        "Stopped - negative reason" = "Negative",
        "All phase III+" = "Phase III+"
    )) %>%
    mutate(comparison = str_replace(comparison, "l2g_", ">")) %>%
    ggplot(aes(x = comparison, y = estimate, color = prediction)) +
    geom_hline(aes(yintercept = 1),
        size = .25,
        linetype = "dashed"
    ) +
    geom_errorbar(aes(ymin = conf.low, ymax = conf.high),
    width = 0) +
    geom_point() +
    scale_color_discrete(name = "Clinical Studies") +
    scale_y_log10(name = "Odds Ratio") +
    scale_x_discrete(name = "GWAS predicted causality (max L2G score)") +
    theme_cowplot(font_size = 12) +
    panel_border(color = "black") +
    theme(
        # strip.text.y = element_blank(),
        strip.background = element_rect(fill = NA),
        panel.grid.major.y = element_blank(),
        legend.position = c(0, 0),
        legend.margin = margin(l = 10, b = 10, unit = "pt"),
        legend.justification = c(0, 0)
    )
p_efficacy_gwas_l2g_score
```